ingestion/src/metadata/ingestion/source/dashboard/powerbi/metadata.py

```diff
@@ -622,13 +624,29 @@ def _fetch_report_from_workspace(
         Method to search the report using id in the workspace dict
         """
         if report_id:
-            report_data = next(
-                (
-                    report
-                    for report in self.context.workspace.reports or []
-                    if report.id == report_id
-                ),
-                None,
-            )
-            return report_data
+            for workspace in self.workspace_data or []:
+                report_data = next(
+                    (
+                        report
+                        for report in workspace.reports or []
+                        if report.id == report_id
+                    ),
+                    None,
+                )
+                return report_data
+        return None
+
+    def _fetch_dataset_workspace(self, dataset_id: Optional[str]) -> Optional[str]:
+        """
+        Method to search the workspace name in which the dataset in contained
+        """
+        if dataset_id:
+            workspace_names = [
+                workspace.name
+                for workspace in self.workspace_data
+                for dataset in workspace.datasets
+                if dataset.id == dataset_id
+            ]
+            return next(iter(workspace_names), None)
```

Comment:
nit - Instead of having a list `workspace_names = [...]` you could just use a generator expression here, just change the `[]` for `()`